From 320af6bbc9d71e3e67cf1d6e59d4a5094324d41c Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 27 Jul 2023 17:08:50 +0100
Subject: [PATCH] curl: Fix call to update_times

This was called in the wrong place so we didn't count the cost of
making the initial HEAD call on each handle.

Fixes: commit 68dddbeb584fb9385915846d259563f74338ffe8
(cherry picked from commit f2d7041f94af15b158c92a96b9d9fdf4d3b7cdef)
---
 plugins/curl/pool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/curl/pool.c b/plugins/curl/pool.c
index 10f9011d..8db69a71 100644
--- a/plugins/curl/pool.c
+++ b/plugins/curl/pool.c
@@ -471,11 +471,11 @@ get_content_length_accept_range (struct curl_handle *ch)
   curl_easy_setopt (ch->c, CURLOPT_HEADERFUNCTION, header_cb);
   curl_easy_setopt (ch->c, CURLOPT_HEADERDATA, ch);
   r = curl_easy_perform (ch->c);
+  update_times (ch->c);
   if (r != CURLE_OK) {
     display_curl_error (ch, r,
                         "problem doing HEAD request to fetch size of URL [%s]",
                         url);
-  update_times (ch->c);
 
     /* Get the HTTP status code, if available. */
     r = curl_easy_getinfo (ch->c, CURLINFO_RESPONSE_CODE, &code);
-- 
2.39.3

